cupy/_core/_routines_manipulation.pyx

```diff
@@ -133,15 +133,16 @@ cdef ndarray _ndarray_ravel(ndarray self, order):
     shape.push_back(self.size)
 
     order_char = internal._normalize_order(order, True)
+    enforce_copy = not self._f_contiguous and not self._c_contiguous
     if order_char == b'A':
         if self._f_contiguous and not self._c_contiguous:
             order_char = b'F'
         else:
             order_char = b'C'
     if order_char == b'C':
-        return _reshape(self, shape)
+        return _reshape(self, shape, enforce_copy)
```

Comment:
The input is copied only if it's required.  Maybe it should have been tested that
```
>>> x = xp.ones((2, 3)); x.ravel().base is x
True
```